certification/engine/engine.go

```diff
@@ -30,33 +30,36 @@ type CheckEngine interface {
 	Results(context.Context) runtime.Results
 }
 
-func NewForConfig(ctx context.Context, cfg certification.Config) (CheckEngine, error) {
-	checks, err := initializeChecks(ctx, cfg.Policy(), cfg)
-	if err != nil {
-		return nil, fmt.Errorf("error initializing checks: %v", err)
-	}
-
-	engine := &internal.CraneEngine{
-		Config:    cfg,
-		Image:     cfg.Image(),
-		Checks:    checks,
-		IsBundle:  cfg.IsBundle(),
-		IsScratch: cfg.IsScratch(),
-	}
+func New(ctx context.Context,
+	image string,
+	checks []certification.Check,
+	dockerconfig string,
+	isBundle,
+	isScratch bool,
+) (CheckEngine, error) {
+	return &internal.CraneEngine{
+		DockerConfig: dockerconfig,
+		Image:        image,
+		Checks:       checks,
+		IsBundle:     isBundle,
+		IsScratch:    isScratch,
+	}, nil
+}
 
-	return engine, nil
+// OperatorCheckConfig contains configuration relevant to an individual check's execution.
+type OperatorCheckConfig struct {
+	ScorecardImage, ScorecardWaitTime, ScorecardNamespace, ScorecardServiceAccount string
+	IndexImage, DockerConfig, Channel, Kubeconfig                                  string
 }
```

Comment:
[Note To Reviewers]: This is an example of the smaller task-focused configuration that's needed to initialize checks. This is what the checks need today, and we can modify this if checks change (as opposed to modifying the larger configs). This is possible because `InitializeChecks` has been replaced by individual policy-based initializing logic.